src/transactions/transactionBuilder/transactionBuilder.ts

```diff
@@ -170,6 +175,54 @@ export function generateTransactionPayloadWithABI(
   return new TransactionPayloadEntryFunction(entryFunctionPayload);
 }
 
+export async function generateViewFunctionPayload(args: InputViewFunctionDataWithRemoteABI): Promise<EntryFunction> {
+  const { moduleAddress, moduleName, functionName } = getFunctionParts(args.function);
+
+  let functionAbi = args.abi;
+  // We fetch the entry function ABI, and then pretend that we already had the ABI
+  if (!functionAbi) {
```

Comment:
code duplication, lets have a function for that

Reply:
They are slightly different, but sure